### ISSUE TITLE
improve UX on remote repositories

### ISSUE THREAD

**Issue by romain-dartigues** [gitahead/gitahead#255]
Enhance the use experience on remote repositories.

For example, I connected GitAhead to :
- a GitLab.com account, where it fetched 33 repositories
- a private GitLab server, where it fetched 15 repositories
- a GitHub.com account, where it fetched 378 repositories

It is really painful to find/switch to a repository in the lateral bar.
I would at least expect something like a "filter list", and group by path (GitHub and GitLab provide at least one "group" level separated by "/").

The "Repository Chooser" does not provide much help (lacking a filter as noted above), and long repository names getting truncated.

![screenshot](https://user-images.githubusercontent.com/393859/63698820-f3b86480-c81f-11e9-81e3-bcfbc606c83d.png)

A shortcut like "jump to repository" which would filter on name by regular expression would be great for me.

**Comment by ftab** [comment]
Grouping by namespaces would be a big help. The [way I'm structuring my self-hosted GitLab repositories](https://gitlab.com/gitlab-org/gitlab/issues/33251#note_243757385), I'm making heavy use of subgroups, with a "project" actually being a subgroup that contains 2 or 3 repositories. Different subgroups will wind up having the same names for some of the repos; hence, without full path view that sidebar is even harder to work with

**Comment by paketep** [comment]
Yeah, this has been my first impression after installing as well (and not a good one).

I have about ~100 repositories in gitlab. If I can't filter them by name, **at the very least** they should be sorted alphabetically, so I don't have to squint through them trying to find the one I'm looking for.

I'm thinking of not going any further, can't believe what OP must have thought having >400.
